### PR TITLE
docs: mention container probes

### LINKED ISSUE #28955: Mention container probes in https://k8s.io/docs/concepts/workloads/pods/
**This is a Feature Request**

**What would you like to be added**
The [Pod](https://kubernetes.io/docs/concepts/workloads/pods/) concept should mention [container probes](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#container-probes).

**Why is this needed**

- Make sure that a reader can find “probe” in https://kubernetes.io/docs/concepts/workloads/pods/ if they search-in-page
- Give readers a clue that this container-level health checking is possible in Kubernetes

**Comments**
A brief mention and a hyperlink to https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#container-probes would be enough. Maybe try to get the keyword “health” in their too.

/language en
/kind feature
/help

### PR DESCRIPTION
**Pull request by slayer321**
/language en
/kind feature

Mention container probes in https://k8s.io/docs/concepts/workloads/pods/ 
closes #28955

### CODE REVIEW

**Comment by reylejano** [comment]
Container probes are already mentioned on the [/docs/concepts/workloads/pods/pod-lifecycle](/docs/concepts/workloads/pods/pod-lifecycle) page with appropriate reference links

**Comment by slayer321** [comment]
Did I misinterpret what was the issue about?
because @sftim said that "The Pod concept should mention container probes".

**Comment by reylejano** [comment]
@slayer321 You're right. Disregard my previous comment.

File: content/en/docs/concepts/workloads/pods/_index.md [hunk 1]
```diff
@@ -282,6 +282,16 @@ on the Kubernetes API server for each static Pod.
...
 
+## Container probes
+
+A Probe is a diagnostic performed periodically by the kubelet on a Container. To perform a diagnostic, the kubelet calls a Handler implemented by the container. There are three types of handlers:
+
+- ExecAction
+- TCPSocketAction
+- HTTPGetAction
+
+learn more about it [here](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#container-probes)
```
Comment (sftim) [inline content/en/docs/concepts/workloads/pods/_index.md:293]:
```suggestion
You can read more about [probes](/docs/concepts/workloads/pods/pod-lifecycle/#container-probes)
in the Pod Lifecycle documentation.
```

File: content/en/docs/concepts/workloads/pods/_index.md [hunk 2]
```diff
@@ -282,6 +282,16 @@ on the Kubernetes API server for each static Pod.
 This means that the Pods running on a node are visible on the API server,
 but cannot be controlled from there.
 
+## Container probes
+
+A Probe is a diagnostic performed periodically by the kubelet on a Container. To perform a diagnostic, the kubelet calls a Handler implemented by the container. There are three types of handlers:
```
Comment (sftim) [inline content/en/docs/concepts/workloads/pods/_index.md:287]: Rule of thumb: if it's in the API but it's not listed by `kubectl api-resources`, either write it in backticks or in sentence case (not UpperCamelCase). Also, use italics when introducing new jargon.
So:
```suggestion
A _probe_ is a diagnostic performed periodically by the kubelet on a container. To perform a diagnostic, the kubelet can invoke different actions:
```

File: content/en/docs/concepts/workloads/pods/_index.md [hunk 3]
```diff
@@ -282,6 +282,16 @@ on the Kubernetes API server for each static Pod.
...
+
+- ExecAction
+- TCPSocketAction
+- HTTPGetAction
```
Comment (sftim) [inline content/en/docs/concepts/workloads/pods/_index.md:291]: How about:
```suggestion
- `ExecAction` (performed with the help of the container runtime)
- `TCPSocketAction` (checked directly by the kubelet)
- `HTTPGetAction` (checked directly by the kubelet)
```
?

**Review by sftim — COMMENTED** [review]
Thanks for this @slayer321 

/lgtm

I'd made some style suggestions; what do you think about these?

File: content/en/docs/concepts/workloads/pods/_index.md (hunk 3 above)
Comment (slayer321) [reply content/en/docs/concepts/workloads/pods/_index.md]: Looks perfect. I have added the suggested changes.

File: content/en/docs/concepts/workloads/pods/_index.md (hunk 2 above)
Comment (slayer321) [reply content/en/docs/concepts/workloads/pods/_index.md]: Nice !! Learned new thing :)

**Review by sftim — COMMENTED** [review]
/lgtm

File: content/en/docs/concepts/workloads/pods/_index.md [hunk 4]
```diff
@@ -282,6 +282,16 @@ on the Kubernetes API server for each static Pod.
...
 
+## Container probes
+
+A _probe_ is a diagnostic performed periodically by the kubelet on a container. To perform a diagnostic, the kubelet can invoke different actions:
+
+- `ExecAction` (performed with the help of the container runtime)
+- `TCPSocketAction` (checked directly by the kubelet)
+- `HTTPGetAction` (checked directly by the kubelet)
+
+You can read more about [probes](docs/concepts/workloads/pods/pod-lifecycle/#container-probes) in the Pod Lifecycle documentation.
```
Comment (tengqm) [inline content/en/docs/concepts/workloads/pods/_index.md:293]:
```suggestion
You can read more about [probes](/docs/concepts/workloads/pods/pod-lifecycle/#container-probes)
in the Pod lifecycle documentation.
```

**Pull request merged**